dm/pkg/schema/tracker.go

```diff
@@ -233,22 +242,39 @@ func NewTracker(
 	// exist by default. So we need to drop it first.
 	err = dom.DDL().DropSchema(se, model.NewCIStr("test"))
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	// init downstreamTracker
 	dsTracker := &downstreamTracker{
 		downstreamConn: downstreamConn,
 		tableInfos:     make(map[string]*DownstreamTableInfo),
 	}
+	tr.Lock()
+	defer tr.Unlock()
+	tr.storePath = storePath
+	tr.store = store
+	tr.dom = dom
+	tr.se = se
+	tr.dsTracker = dsTracker
+	tr.closed.Store(false)
+	return nil
+}
 
-	return &Tracker{
-		storePath: storePath,
-		store:     store,
-		dom:       dom,
-		se:        se,
-		dsTracker: dsTracker,
-	}, nil
+// NewTracker creates a new tracker. It's preserved for test.
+func NewTracker(
```

Comment:
prefer to name `NewDumpTracker` as `NewTracker`, and remove this one.

for test, we change them to `Init`

Reply:
Maybe leaving a function invoking `NewTracker` and subsequently `Init` is better, so as to make the test more succinct.

Reply:
then name it explicitly as something like NewTrackerForTest. `NewTracker` is how we name normal ctor

Reply:
IDE should be able to easily rename them all

Reply:
8dceb81